File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        write = 0                   # write pointer for where to place next valid element

        # iterate through the array
        for num in nums:
            # if less than 2 elements written, always write
            # if num > nums[write -2], it means this is at most the second occurrence
            if write < 2 or num > nums[write - 2]:
                nums[write] = num               # write valid value
                write += 1                      # increment write pointer

        # the first 'write' elements are the answer
        return write

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        if (nums.Length <= 2)
            return nums.Length;

        int left = 2;

        for (int right = 2; right < nums.Length; right++) {
            if (nums[right] != nums[left - 2]) {
                nums[left] = nums[right];
                left++;
            }
        }

        return left;
    }
}